flake8_trio.py

```diff
@@ -200,6 +205,36 @@ def has_decorator(decorator_list: List[ast.expr], *names: str):
     return False
 
 
+def regex_has_decorator(decorator_list: List[ast.expr], *names: str):
+    def wild_match(match_str: str, target: str) -> bool:
+        if match_str == "*":
+            return True
+        if "*" not in match_str:
+            return match_str == target
+        return bool(re.match(re.escape(match_str).replace("\\*", ".*"), target))
```

Comment:
Seems much safer to construct the fully-qualified decorator name as a string, then [use the `fnmatch` module](https://docs.python.org/3/library/fnmatch.html#fnmatch.fnmatchcase) for explicitly glob-style matching.  This more complicated logic looks like an invitation to weird edge cases.

Reply:
hmm, that does not support differentiating `.` - so `mydeco*` would match `mydecoratormodule.foo`.
But it should be possible to get rid of the recursion and just do a `re.match` against a constructed string, which should make for simpler logic.

Reply:
I can see where you're coming from, but IMO the simplicity of "just use fnmatch" (both for implementation, and documentation) outweighs the slightly more precise but considerably harder to explain regex and special handling approach.

Reply:
After playing with both I wasn't convinced at first and got the regex code simplified and readable, but then I realized that saying "ignore anything in module `x` or any of it's sub-packages" wasn't possible with simple regex logic and settled on fnmatch.